monolithic_integration_test/tests/janus.rs

```diff
@@ -1,39 +1,163 @@
+    leader_task: Task,
+    /// The private key corresponding to the collector HPKE configuration in the task configured in
+    /// both Janus aggregators.
+    collector_private_key: HpkePrivateKey,
+
+    /// Handle to the leader's resources, which are released on drop.
+    _leader: Janus,
+    /// Handle to the helper's resources, which are released on drop.
+    _helper: Janus,
+}
+
+impl JanusPair {
+    fn in_cluster_aggregator_url(namespace: &str) -> Url {
+        Url::parse(&format!(
+            "http://aggregator.{namespace}.svc.cluster.local:80"
+        ))
+        .unwrap()
+    }
+
+    /// Set up a new Janus test instance with the provided task. If the environment variables listed
+    /// below are set, then the test instance is abcked by an already-running Kubernetes cluster.
+    /// Otherwise, an ephemeral, in-process instance of Janus is spawned. In either case, the Janus
+    /// aggregators' API endpoints will be available on the local loopback interface, at the URLs in
+    /// `Self.leader_task.aggregator_endpoints`.
+    ///
+    /// If connecting to the Kubernetes cluster API (k8s API, not DAP API) at an IP address over
+    /// HTTPS (e.g., "https://127.0.0.1:42356"), then the `monolithic_integration_test` package must
+    /// be built with the `kube-openssl` feature, as the default rustls can't validate IP addresses
+    /// in certificates.
+    /// e.g., `cargo test --features kube-openssl --packagep monolithic_integration_test`
+    ///
+    /// Environment variables:
+    ///  - `JANUS_E2E_KUBE_CONFIG_PATH`: The path to a `kubectl` configuration file containing
+    ///    the information needed to connect to the Kubernetes cluster where the test is to be run.
+    ///  - `JANUS_E2E_KUBECTL_CONTEXT_NAME`: The name of a `kubectl` context defined in the config
+    ///    at `JANUS_E2E_KUBE_CONFIG_PATH`. The context will be used to connect to the Kubernetes
+    ///    cluster where the test is to be run. The context should grant sufficient cluster
+    ///    permissions to view secrets and forward ports to services.
+    ///  - `JANUS_E2E_LEADER_NAMESPACE`: The Kubernetes namespace where the DAP leader is deployed.
+    ///  - `JANUS_E2E_HELPER_NAMESPACE`: The Kubernetes namespace where the DAP helper is deployed.
+    pub async fn new() -> Self {
+        let (leader_port, helper_port) = pick_two_unused_ports();
+        let (collector_hpke_config, collector_private_key) =
+            generate_test_hpke_config_and_private_key();
+        let (mut leader_task, mut helper_task) =
+            create_test_tasks(leader_port, helper_port, &collector_hpke_config);
+
+        let kubeconfig_path = env::var("JANUS_E2E_KUBE_CONFIG_PATH");
+        let kubectl_context_name = env::var("JANUS_E2E_KUBECTL_CONTEXT_NAME");
+        let leader_namespace = env::var("JANUS_E2E_LEADER_NAMESPACE");
+        let helper_namespace = env::var("JANUS_E2E_HELPER_NAMESPACE");
```

Comment:
My thinking here is that developers at their workstations will do `cargo test` and run the tests against the in-process Janus, as will `janus` CI. But in `janus-ops`, we'll check out `janus`, set all these environment variables based on the Kind cluster that CI sets up, and run `cargo --package monolithic_integration_test --features kube-openssl`. However I'm open to other ways to solve this. We could also expose a feature in `monolithic_integration_test` that governs which way the test is run.

Reply:
I prefer this environment variable solution over a feature.

Reply:
Same: I think we'll need to specify the data in the env vars somehow, so even if we went with a feature, we'd still need an env var (or moral equivalent) to pass along the connection data.